gp-includes/advanced-permissions.php

```diff
@@ -42,14 +46,48 @@ function gp_route_translation_set_permissions_to_validator_permissions( $verdict
 }
 
 function gp_allow_everyone_to_translate( $verdict, $args ) {
+	if ( is_bool( $verdict ) ) {
+		return $verdict;
+	}
+
 	if ( 'edit' == $args['action'] && 'translation-set' == $args['object_type'] ) {
 		return is_user_logged_in();
 	}
 
 	return $verdict;
 }
 
+/**
+ * Maps the translation check to the translation-set.
+ *
+ * @since 2.3.0
+ *
+ * @param string|bool $verdict Previous decision whether the user can do this.
+ * @param array       $args    Permission details.
+ * @return string|bool New decision whether the user can do this.
+ */
+function gp_allow_approving_translations_with_validator_permissions( $verdict, $args ) {
+	if ( is_bool( $verdict ) ) {
+		return $verdict;
+	}
+
+	if ( 'approve' === $args['action'] && 'translation' === $args['object_type'] ) {
+		$args['object_type'] = 'translation-set';
+
+		if ( isset( $args['extra']['translation']->translation_set_id ) ) {
+			$args['object_id'] = $args['extra']['translation']->translation_set_id;
+		} else {
+			return $verdict;
+		}
+
+		return gp_route_translation_set_permissions_to_validator_permissions( $verdict, $args );
+	}
+
+	return $verdict;
+}
+
 add_filter( 'gp_can_user', 'gp_recurse_project_permissions', 10, 2 );
 add_filter( 'gp_can_user', 'gp_recurse_validator_permission', 10, 2 );
 add_filter( 'gp_pre_can_user', 'gp_route_translation_set_permissions_to_validator_permissions', 10, 2 );
 add_filter( 'gp_pre_can_user', 'gp_allow_everyone_to_translate', 10, 2 );
+add_filter( 'gp_pre_can_user', 'gp_allow_approving_translations_with_validator_permissions', 9, 2 );
```

Comment:
Yeah, I'd like to see all the default hooks using the default priority, if possible.